Siamese Network(Task 7) s4636418 Yicheng Jia

YichengJia opened the pull request:
Hi Teaching Teams,
I got an extension until today so please don't mark me as late submission.
Thank you!

YichengJia committed:
Create readme.md

YichengJia committed:
Add initial framework code files 

These files are the basic framework code for implementing the requirements of this task.

YichengJia committed:
Add initial framework code files

These files are the basic framework code for implementing the requirements of this task.

YichengJia committed:
Delete dataset.py

YichengJia committed:
Delete modules.py

YichengJia committed:
Delete predict.py

YichengJia committed:
Delete train.py

YichengJia committed:
Update readme.md

YichengJia committed:
Update some details.

YichengJia committed:
Bugs & Errors Fixed

Now I have fixed bugs in modules.py, dataset.py and train.py. The epoch is only 10 since I just want to make sure there is no any error when run train.py.

YichengJia committed:
Update readme.md

Add AD_NC directory structure schematic.

YichengJia committed:
Update dataset.py

YichengJia committed:
Update modules, transforms, optimzer and scheduler

The learning rate is always around 50% and I'm trying my best to fix it.

YichengJia committed:
Update readme.md

YichengJia committed:
Update readme.md

YichengJia committed:
Update readme.md

YichengJia committed:
Update readme.md

YichengJia committed:
Update dataset.py

YichengJia committed:
Update modules.py

YichengJia committed:
Update train.py

YichengJia committed:
Update predict.py

YichengJia committed:
Update train.py

YichengJia committed:
Update dataset.py

YichengJia committed:
Update modules.py

YichengJia committed:
Update train.py

YichengJia committed:
Update modules.py

YichengJia committed:
Update dataset.py

YichengJia committed:
Update train.py

YichengJia committed:
Update modules.py

YichengJia committed:
Update dataset.py

YichengJia committed:
Update train.py

YichengJia committed:
Update predict.py

YichengJia committed:
Finalise readme.md

YichengJia committed:
Finalise readme.md

YichengJia committed:
Finalise dataset.py

YichengJia committed:
Finalise dataset.py

YichengJia committed:
Finalise modules.py

YichengJia committed:
Finalise train.py

YichengJia committed:
Finalise train.py

YichengJia committed:
Finalise modules.py

YichengJia committed:
Finalise dataset.py

YichengJia committed:
Update train.py

YichengJia committed:
Update predict.py

YichengJia committed:
Update readme.md

YichengJia committed:
Create readme

YichengJia committed:
Add output images

These files are output.
Files with '_' are generated via tensorboard.
Files without '_' are generated via pyplot, these datas are downloaded from server due to the limit of my hardware.

YichengJia committed:
Add files via upload

YichengJia committed:
Update readme.md

YichengJia committed:
Update readme.md

YichengJia committed:
Finalise predict.py

YichengJia committed:
Finalise modules.py

YichengJia committed:
Finalise dataset.py

YichengJia committed:
Finalise train.py

YichengJia committed:
Add files via upload

YichengJia committed:
Update readme.md

YichengJia committed:
Create readme

YichengJia committed:
Delete recognition/Siamese Network_Yicheng Jia/Best_Modules directory

YichengJia committed:
Update readme.md

nathasha-naranpanawa commented:
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:  Good
- Clear flow of information
- Algortihm and usage explained well
- Some plots provided

Commit messages:  Poor, minimal and not detailed

Code: 
- design is OK
- well commented
- doesn't seem to have performed data augmentation or any hyper-parameter tuning

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification**
- achieves good accuracy for similarity detection with the siamese 

General comments:
- **algorithm solves the respective problem only half-way**
- you need to have added a classifier that can classify an individual image as AD or NC with their features extracted by the trained siamese network

YichengJia committed:
Fix the directory structure

YichengJia committed:
Delete recognition/Siamese Network_Yicheng Jia/Images/readme

YichengJia commented:
Hi Course Team,
I have modified my codes and fixed the structure. Please check my new codes later.
Thank you very much for any help you provided this semester!

YichengJia committed:
Update the structure

YichengJia committed:
Added classifier

YichengJia committed:
Added classifier

YichengJia committed:
Added actions when handle classifier

YichengJia committed:
Added train_classifier function to create a classifier and show results

YichengJia committed:
Add the screenshot result of classifer

YichengJia committed:
Add the output of classifier

YichengJia committed:
Added output of classifier

YichengJia committed:
Added output of classifier

shakes76 commented:
# Marking
Marked as per the extended due date and changes after which aren't necessarily allowed to contribute to grade for fairness
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Partially Solves problem no classification in initial PR -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no augmentation -1
Difficulty: Hard
## Commit Log
Meaningful commit messages minimal -1
Progressive commits used
## Documentation
ReadMe acceptable, no refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required, done already -2 +2
Request Description good
